boards/rp-pico/examples/pico_display_pack_2.rs

```diff
@@ -0,0 +1,170 @@
+//!
+//! It will demonstrate how to draw on the screen using the embedded_graphics API
+//!
+//! See the `Cargo.toml` file for Copyright and license details.
+
+#![no_std]
+#![no_main]
+
+// The macro for our start-up function
+use rp_pico::entry;
+
+// Time handling traits
+use embedded_time::rate::*;
+
+// Ensure we halt the program on panic (if we don't mention this crate it won't
+// be linked)
+use panic_halt as _;
+
+// Pull in any important traits
+use rp_pico::hal::prelude::*;
+
+// A shorter alias for the Peripheral Access Crate, which provides low-level
+// register access
+use rp_pico::hal::pac;
+
+// A shorter alias for the Hardware Abstraction Layer, which provides
+// higher-level drivers.
+use rp_pico::hal;
+
+// ST7789 driver
+use st7789::{Orientation, ST7789};
+
+// SPI display interface
+use display_interface_spi::SPIInterface;
+
+// Graphics drawing utilities
+use embedded_graphics::pixelcolor::Rgb565;
+use embedded_graphics::prelude::*;
+use embedded_graphics::primitives::*;
+
+// GPIO traits
+use embedded_hal::digital::v2::OutputPin;
+use embedded_hal::PwmPin;
+
+/// External high-speed crystal on the Raspberry Pi Pico board is 12 MHz. Adjust
+/// if your board has a different frequency
+const XTAL_FREQ_HZ: u32 = 12_000_000u32;
+
+#[entry]
+fn main() -> ! {
+    // Grab our singleton objects
+    let mut pac = pac::Peripherals::take().unwrap();
+    let core = pac::CorePeripherals::take().unwrap();
+
+    // Set up the watchdog driver - needed by the clock setup code
+    let mut watchdog = hal::Watchdog::new(pac.WATCHDOG);
+
+    // Configure the clocks
+    let clocks = hal::clocks::init_clocks_and_plls(
+        XTAL_FREQ_HZ,
+        pac.XOSC,
+        pac.CLOCKS,
+        pac.PLL_SYS,
+        pac.PLL_USB,
+        &mut pac.RESETS,
+        &mut watchdog,
+    )
+    .ok()
+    .unwrap();
+
+    let mut delay = cortex_m::delay::Delay::new(core.SYST, clocks.system_clock.freq().integer());
+
+    // The single-cycle I/O block controls our GPIO pins
+    let sio = hal::Sio::new(pac.SIO);
+
+    // Set the pins up according to their function on this particular board
+    let pins = rp_pico::Pins::new(
+        pac.IO_BANK0,
+        pac.PADS_BANK0,
+        sio.gpio_bank0,
+        &mut pac.RESETS,
+    );
+
+    // Init PWMs
+    let mut pwm_slices = hal::pwm::Slices::new(pac.PWM, &mut pac.RESETS);
+
+    // Configure and enable PWMs
+    let pwm3 = &mut pwm_slices.pwm3;
+    pwm3.set_ph_correct();
+    pwm3.enable();
+    let pwm4 = &mut pwm_slices.pwm4;
+    pwm4.set_ph_correct();
+    pwm4.enable();
+
+    // Configure button inputs
+    // 12 - A button
+    let _a_pin = pins.gpio12.into_floating_input();
+    // 13 - B button
+    let _b_pin = pins.gpio13.into_floating_input();
+    // 14 - X button
+    let _x_pin = pins.gpio14.into_floating_input();
+    // 15 - Y button
+    let _y_pin = pins.gpio15.into_floating_input();
```

Comment:
These switches do not have external pull-ups, you should enable the internal ones or you'll never be able to detect a button press.
```suggestion
    // Configure button inputs
    // The push-buttons connect their respective pin to ground, this is often referred to as "active-low"
    // Since the board does not have external pull-up resistors, enable internal pull-up on each input
    // 12 - A button
    let _a_pin = pins.gpio12.into_pull_up_input();
    // 13 - B button
    let _b_pin = pins.gpio13.into_pull_up_input();
    // 14 - X button
    let _x_pin = pins.gpio14.into_pull_up_input();
    // 15 - Y button
    let _y_pin = pins.gpio15.into_pull_up_input();
```